ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -389,8 +438,14 @@ private static TableRows queryRowsLocally(
   private static TableRows forwardTo(
       final KsqlNode owner,
       final ConfiguredStatement<Query> statement,
-      final ServiceContext serviceContext
+      final ServiceContext serviceContext,
+      final PullQueryContext pullQueryContext
   ) {
+    // Rewrite the expression to only query for the particular keys we care about for this node.
+    // Otherwise, we'll risk reading standby data for other partitions.
```

Comment:
I'm worried this approach doesn't scale well for things other than `IN` queries (not to mention that it feels hacky). Instead, it probably makes sense to have a way to specify that a pull query (internally routed only) should only read from certain partitions. Otherwise, how would we handle things like range queries? I can't "rewrite" the range query and avoid the risk of reading standby data. 

Generally, I think it might make sense to think about communicating pull queries internally using something other than just the SQL statement.

Reply:
Yeah, that's fair.  We actually were discussing range queries as well and it's probably easier to use a unified approach.  I think that it should be sufficient to specify partition when reading from the table (to ensure you're only reading from the actives and standbys you intend) and also on the second hop, avoid reading keys that are not hosted on this machine or that don't agree with those specified partitions.  That same approach will work on range queries as well.

Reply:
Ok, implemented that other approach where partition is specified.  No rewriting necessary.